core/src/main/java/hudson/Functions.java

```diff
@@ -1156,6 +1157,43 @@ public static String getFooterURL() {
                 Jenkins.get().hasPermission(d.getRequiredGlobalConfigPagePermission()) || Jenkins.get().hasPermission(Jenkins.SYSTEM_READ)));
     }
 
+    /**
+     * Checks if the current security principal has one of the supplied permissions.
+     *
+     * @since TODO
+     */
+    public static boolean hasAnyPermission(AccessControlled ac, Permission[] permissions) {
+        if (permissions == null || permissions.length == 0) {
+            return true;
```

Comment:
This behavior is shitty and broken, but if we're going to address it in `#hasPermission`, we can also fix up this.